### PR TITLE
Readme updates

### PR DESCRIPTION
**Pull request by stockholmux**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Does `npm test` pass with this change (including linting)?
- [ ] Is the new or changed code fully tested?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change
Refined the `createClient` section of the readme. Fixed typos, made the language consistent, removed some bits of opinion, and switched from a bullet list to a md table

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+__Tip:__ If the Redis server runs on the same machine as the client consider using unix sockets if possible to increase throughput.
+
+#### `options` object properties
+| Property  | Default   | Description |
+|-----------|-----------|-------------|
+| host      | 127.0.0.1 | IP address of the Redis server |
+| port      | 6379      | Port of the Redis server |
+| path      | null      | The UNIX socket string of the Redis server |
+| url       | null      | The URL of the Redis server. Uses the format `[redis:]//[user][:password@][host][:port][/db-number][?db=db-number[&password=bar[&option=value]]]` More info avaliable at [IANA](http://www.iana.org/assignments/uri-schemes/prov/redis) |
```
Comment (BridgeAR) [inline README.md]: 1. To shorten this you might change the `Uses the format` to `Format: `
2. Dot missing after the URL shema

File: README.md [hunk 2]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| parser    | hiredis   |  If hiredis is not installed, automatic fallback to the built-in javascript parser |
+| string_numbers | null   | Set to `true`, `node_redis will return Redis number values as Strings instead of javascript Numbers. Useful if you need to handle big numbers (above `Number.MAX_SAFE_INTEGER === 2^53`). Hiredis is incapable of this behavior, so setting this option to `true` will result in the built-in javascript parser being used no matter the value of the `parser` option. |
```
Comment (BridgeAR) [inline README.md]: Missing backtick after `node_redis`

File: README.md [hunk 3]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| return_buffers | false | If set to `true`, then all replies will be sent to callbacks as Buffers instead of Strings. |
+| detect_buffers | false | If set to `true`, then replies will be sent to callbacks as Buffers. This option lets you switch between Buffers and Strings on a per-command basis, whereas `return_buffers` applies to every command on a client. __Note__: This doesn't work properly with the pubsub mode. A subscriber has to either always return strings or buffers. |
```
Comment (BridgeAR) [inline README.md]: ...return `Strings` or `Buffers`.

File: README.md [hunk 4]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| socket_keepalive | true | If set to `true`, the keep-alive functionality is enabled on the underlying socket |
```
Comment (BridgeAR) [inline README.md]: Dot missing at the end

File: README.md [hunk 5]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| no_ready_check | false |  When a connection is established to the Redis server, the server might still be loading the database from disk. While loading the server will not respond to any commands. To work around this, `node_redis` has a "ready check" which sends the `INFO` command to the server. The response from the `INFO` command indicates whether the server is ready for more commands. When ready, `node_redis` emits a `ready` event. Setting `no_ready_check` to `true` will inhibit this check. |
```
Comment (BridgeAR) [inline README.md]: A comma missing after "while loading"

File: README.md [hunk 6]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| enable_offline_queue |  true | By default, if there is no active connection to the Redis server, commands are added to a queue and are executed once the connection has been established. Setting `enable_offline_queue` to `false` will disable this feature and the callback will be executed immediately with an error, or an error will be emitted if no callback is specified. |
+| retry_max_delay | null | By default, every time the client tries to connect and fails the reconnection delay almost doubles. This delay normally grows infinitely, but setting `retry_max_delay` limits it to the maximum value, provided in milliseconds. |
```
Comment (BridgeAR) [inline README.md]: A comma missing after "connect and fails". No comma needed before "provided"

File: README.md [hunk 7]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| connect_timeout | 3600000 | Setting `connect_timeout` limits the total time for the client to connect and reconnect. The value is provided in milliseconds and is counted from the moment a new client is created or when a connection is lost. The last retry is going to happen exactly at the timeout time. Default is to try connecting until the default system socket timeout has been exceeded and to try reconnecting until 1h has elapsed. |
+| max_attempts | 0 | __Deprecated__ _Please use `retry_strategy` instead._ By default client will try reconnecting until connected. Setting `max_attempts` limits total amount of connection tries. Setting this to 1 will prevent any reconnect tries. |
```
Comment (BridgeAR) [inline README.md]: 1. Change "tries" to "attempts"
2. By default, a client...

File: README.md [hunk 8]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| retry_unfulfilled_commands | false | If set to `true`, all commands that were unfulfilled while the connection is lost will be retried after the connection has been reestablished. Use this with caution, if you use state altering commands (e.g. `incr`). This is especially useful if you use blocking commands. |
```
Comment (BridgeAR) [inline README.md]: No comma before "if you use"

File: README.md [hunk 9]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| password | null | If set, client will run Redis auth command on connect. Alias `auth_pass` __Note__ `node_redis` < 2.5 must use `auth_pass` |
+| db | null | If set, client will run Redis `select` command on connect. |
+| family | IPv4 | You can force using IPv6 if you set the family to 'IPv6'. See Node.js [net](https://nodejs.org/api/net.html) or [dns](https://nodejs.org/api/dns.html) modules how to use the family type. |
```
Comment (BridgeAR) [inline README.md]: ...modules __on__ how to use...

File: README.md [hunk 10]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| disable_resubscribing | false | If set to `true`, a client won't resubscribe after disconnecting |
```
Comment (BridgeAR) [inline README.md]: Dot at the end

File: README.md [hunk 11]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| rename_commands | null | Passing a object with renamed commands to use instead of the original functions. See the [Redis security topics](http://redis.io/topics/security) for more info. |
```
Comment (BridgeAR) [inline README.md]: ...a<b>n</b> object...

File: README.md [hunk 12]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| tls | null | An object containing options to pass to [tls.connect](http://nodejs.org/api/tls.html#tls_tls_connect_port_host_options_callback) to set up a TLS connection to Redis (if, for example, it is set up to be accessible via a tunnel). |
+| prefix | null | String to prefix all used keys with that string as prefix (e.g. `namespace:test`). Please be aware, that the `keys` command will not be prefixed. The `keys` command has a "pattern" as argument and no key and it would be impossible to determine the existing keys in Redis if this would be prefixed. |
```
Comment (BridgeAR) [inline README.md]: 1. `A string used to prefix all used keys` to replace the first sentence
2. No comma before "that"

File: README.md [hunk 13]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| retry_strategy | function | Function that receives a options object as parameter including the retry `attempt`, the `total_retry_time` indicating how much time passed since the last time connected, the `error` why the connection was lost and the number of `times_connected` in total. If you return a number from this function, the retry will happen exactly after that time in milliseconds. If you return a non-number no further retry will happen and all offline commands are flushed with errors. Return a error to return that specific error to all offline commands. Example below. |
```
Comment (BridgeAR) [inline README.md]: 1. `A function` in the beginning
2. ...a<b>n</b> options object...
3. A comma after "non-number"
4. ...a<b>n</b> error...
5. `Return an error that is then passed to all offline commands` to replace the last sentence

File: README.md [hunk 14]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
+| enable_offline_queue |  true | By default, if there is no active connection to the Redis server, commands are added to a queue and are executed once the connection has been established. Setting `enable_offline_queue` to `false` will disable this feature and the callback will be executed immediately with an error, or an error will be emitted if no callback is specified. |
+| retry_max_delay | null | By default, every time the client tries to connect and fails the reconnection delay almost doubles. This delay normally grows infinitely, but setting `retry_max_delay` limits it to the maximum value, provided in milliseconds. |
+| connect_timeout | 3600000 | Setting `connect_timeout` limits the total time for the client to connect and reconnect. The value is provided in milliseconds and is counted from the moment a new client is created or when a connection is lost. The last retry is going to happen exactly at the timeout time. Default is to try connecting until the default system socket timeout has been exceeded and to try reconnecting until 1h has elapsed. |
```
Comment (BridgeAR) [inline README.md]: ... created or `from the time the connection is lost`

File: README.md [hunk 15]
```diff
@@ -166,57 +166,42 @@ If false is returned the stream had to buffer.
...
-port and host are probably fine and you don't need to supply any arguments. `createClient()` returns a `RedisClient` object.
+If you have `redis-server` running on the same machine as node, then the defaults for
+port and host are probably fine and you don't need to supply any arguments. `createClient()` returns a `RedisClient` object. Otherwise, `createClient()` accepts these arguments:
 
-If the redis server runs on the same machine as the client consider using unix sockets if possible to increase throughput.
-
-### overloading
 * `redis.createClient([options])`
-* `redis.createClient(unix_socket[, options])`
+* `redis.createClient(unix_socket[, options])`
```
Comment (BridgeAR) [inline README.md]: Trailing whitespace

**Comment by BridgeAR** [comment]
A couple of comments about mistakes I made myself :laughing: 
Otherwise LGTM

**Comment by BridgeAR** [comment]
@stockholmux [merged](https://github.com/NodeRedis/node_redis/commit/0311198f7d08ced994493ee354b8da05673a29d0). Thanks a lot for the contribution :smiley: 

This makes the options definitely better readable.

**Comment by stockholmux** [comment]
Thanks for making those alterations - I got quite busy.

**Pull request closed without merging**